skbio/util/_decorator.py

```diff
@@ -82,3 +85,207 @@ def __get__(self, cls, owner):
 
     def __set__(self, obj, value):
         raise AttributeError("can't set attribute")
+
+
+class _state_decorator(object):
+    """ Base class for decorators of all public functionality.
+    """
+
+    _line_prefix = '\n        '
+    _required_kwargs = []
```

Comment:
I recommend making this a tuple to avoid bugs where this is mutated and it affects all instances since it is a class variable.